enterprise/security/src/test/java/org/slf4j/impl/Neo4jLoggerFactoryTest.java

```diff
@@ -0,0 +1,260 @@
+/*
+ * Copyright (c) 2002-2016 "Neo Technology,"
+ * Network Engine for Objects in Lund AB [http://neotechnology.com]
+ *
+ * This file is part of Neo4j.
+ *
+ * Neo4j is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as
+ * published by the Free Software Foundation, either version 3 of the
+ * License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program. If not, see <http://www.gnu.org/licenses/>.
+ */
+package org.slf4j.impl;
+
+import org.junit.Before;
+import org.junit.Test;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.slf4j.Marker;
+
+import org.neo4j.logging.Log;
+import org.neo4j.logging.LogProvider;
+
+import static org.junit.Assert.assertFalse;
+import static org.junit.Assert.assertTrue;
+import static org.mockito.Matchers.anyString;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.times;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.when;
+
+public class Neo4jLoggerFactoryTest
+{
+    private final String format = "hello %s, %s, %s";
+    private final String arg1 = "1";
+    private final String arg2 = "2";
+    private final String arg3 = "3";
+    private final String[] args = { arg1, arg2, arg3 };
+
+    private final LogProvider neo4jLogProvider = mock( LogProvider.class );
+    private final Log neo4jLog = mock( Log.class );
+    private final Marker marker = mock( Marker.class );
+
+    @Before
+    public void setUp()
+    {
+        when( neo4jLogProvider.getLog( anyString() ) ).thenReturn( neo4jLog );
+    }
+
+    @Test
+    public void shouldSupporIsDebugEnabled()
+    {
+        // Given
+        StaticLoggerBinder.setNeo4jLogProvider( neo4jLogProvider );
+        Logger slf4jLog = LoggerFactory.getLogger( Neo4jLoggerFactoryTest.class );
+        when( neo4jLog.isDebugEnabled() ).thenReturn( true );
+
+        // When
+        boolean debugEnabled = slf4jLog.isDebugEnabled();
+        when( neo4jLog.isDebugEnabled() ).thenReturn( false );
+        boolean debugDisabled = slf4jLog.isDebugEnabled();
+
+        // Then
+        verify( neo4jLog, times(2) ).isDebugEnabled();
+        assertTrue( debugEnabled );
+        assertFalse( debugDisabled );
```

Comment:
What is this test actually testing? The asserted results are given by the mocks; what actual logic is used?